### PR TITLE
Suppress warning CS1998: This async method lacks 'await'

### PR DESCRIPTION
**Pull request by lurumad**
Hi,

I've seen a lot of warnings when you compile the solution related with async/await stuffs "Suppress warning CS1998: This async method lacks 'await'"

If you have nothing to await why not just return Task.FromResult?

Regards!

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Bindings/BlobExtensionConfig.cs [hunk 1]
```diff
@@ -198,13 +198,13 @@ private class MultiBlobContext
...
 
-        private async Task<CloudBlobClient> GetClientAsync(
+        private Task<CloudBlobClient> GetClientAsync(
          BlobAttribute blobAttribute,
          CancellationToken cancellationToken)
         {
             var account = _accountProvider.Get(blobAttribute.Connection, _nameResolver);
             var client = account.CreateCloudBlobClient();
-            return client;
+            return Task.FromResult(client);
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Bindings/BlobExtensionConfig.cs]: This has already been fixed in latest: https://github.com/Azure/azure-webjobs-sdk/blob/3c3fc7dfa107bd65f1ea4e6d03580de66f404695/src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Config/BlobsExtensionConfigProvider.cs#L294

File: src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Triggers/BlobTriggerAttributeBindingProvider.cs [hunk 2]
```diff
@@ -114,14 +114,14 @@ internal class BlobTriggerAttributeBindingProvider : ITriggerBindingProvider
...
-        public async Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext context)
+        public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext context)
         {
             ParameterInfo parameter = context.Parameter;
             var blobTriggerAttribute = TypeUtility.GetResolvedAttribute<BlobTriggerAttribute>(context.Parameter);
 
             if (blobTriggerAttribute == null)
             {
-                return null;
+                return Task.FromResult<ITriggerBinding>(null);
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Triggers/BlobTriggerAttributeBindingProvider.cs]: This has already been fixed in latest: https://github.com/Azure/azure-webjobs-sdk/blob/3c3fc7dfa107bd65f1ea4e6d03580de66f404695/src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Triggers/BlobTriggerAttributeBindingProvider.cs#L84

File: src/Microsoft.Azure.WebJobs.Extensions.Storage/CloudStorageAccount/CloudStorageAccountBindingProvider.cs [hunk 3]
```diff
@@ -19,19 +19,19 @@ public CloudStorageAccountBindingProvider(StorageAccountProvider accountProvider
...
 
-        public async Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
         {
             ParameterInfo parameter = context.Parameter;
 
             if (parameter.ParameterType != typeof(CloudStorageAccount))
             {
-                return null;
+                return Task.FromResult<IBinding>(null);
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Extensions.Storage/CloudStorageAccount/CloudStorageAccountBindingProvider.cs]: Already fixed in latest: https://github.com/Azure/azure-webjobs-sdk/blob/3c3fc7dfa107bd65f1ea4e6d03580de66f404695/src/Microsoft.Azure.WebJobs.Extensions.Storage/CloudStorageAccount/CloudStorageAccountBindingProvider.cs#L33

File: src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Bindings/QueueBindingProvider.cs [hunk 4]
```diff
@@ -201,7 +201,7 @@ internal async Task<CloudQueue> GetQueueAsync(QueueAttribute attrResolved)
                 string queueName = attrResolved.QueueName.ToLowerInvariant();
                 QueueClient.ValidateQueueName(queueName);
 
-                return client.GetQueueReference(queueName);
+                return Task.FromResult(client.GetQueueReference(queueName));
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Bindings/QueueBindingProvider.cs]: Fixed in latest: https://github.com/Azure/azure-webjobs-sdk/blob/ef265e20d4357f2a82871b63a778c99cb6aac8db/src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Config/QueuesExtensionConfigProvider.cs#L205

File: src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Triggers/QueueTriggerAttributeBindingProvider.cs [hunk 5]
```diff
@@ -45,14 +45,14 @@ internal class QueueTriggerAttributeBindingProvider : ITriggerBindingProvider
...
-        public async Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext context)
+        public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext context)
         {
             ParameterInfo parameter = context.Parameter;
             var queueTrigger = TypeUtility.GetResolvedAttribute<QueueTriggerAttribute>(context.Parameter);
 
             if (queueTrigger == null)
             {
-                return null;
+                return Task.FromResult<ITriggerBinding>(null);
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Triggers/QueueTriggerAttributeBindingProvider.cs]: Fixed in latest: https://github.com/Azure/azure-webjobs-sdk/blob/3c3fc7dfa107bd65f1ea4e6d03580de66f404695/src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Triggers/QueueTriggerAttributeBindingProvider.cs#L82

File: src/Microsoft.Azure.WebJobs.ServiceBus/Triggers/MessageValueProvider.cs [hunk 6]
```diff
@@ -67,14 +67,14 @@ public string ToInvokeString()
...
         }
 
-        private static async Task<string> GetBase64StringAsync(Message clonedMessage,
+        private static Task<string> GetBase64StringAsync(Message clonedMessage,
             CancellationToken cancellationToken)
         {
             if (clonedMessage.Body == null)
             {
-                return null;
+                return Task.FromResult<string>(null);
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.ServiceBus/Triggers/MessageValueProvider.cs]: Fixed in latest: https://github.com/Azure/azure-webjobs-sdk/blob/3c3fc7dfa107bd65f1ea4e6d03580de66f404695/src/Microsoft.Azure.WebJobs.ServiceBus/Triggers/MessageValueProvider.cs#L76

**Comment by mathewc** [comment]
@lurumad Thanks for noticing and sending the PR! We didn't see your PR but noticed these issues ourselves recently and they've been fixed already - thanks again! This can probably be closed.

**Comment by lurumad** [comment]
It is my pleasure to contribute to this repo!

Regards!

**Pull request closed without merging**